### PR TITLE
feature: Added Plan Limits

### PR DESCRIPTION
**Pull request by Urento**
Docs: https://docs.gitlab.com/ee/api/plan_limits.html

### CODE REVIEW

File: plan_limits.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+type PlanLimit struct {
+	PlanName                   *string `json:"plan_name,omitempty"`
+	ConanMaxFileSize           *int    `json:"conan_max_file_size,omitempty"`
+	GenericPackagesMaxFileSize *int    `json:"generic_packages_max_file_size,omitempty"`
+	HelmMaxFileSize            *int    `json:"helm_max_file_size,omitempty"`
+	MavenMaxFileSize           *int    `json:"maven_max_file_size,omitempty"`
+	NpmMaxFileSize             *int    `json:"npm_max_file_size,omitempty"`
+	NugetMaxFileSize           *int    `json:"nuget_max_file_size,omitempty"`
+	PyPiMaxFileSize            *int    `json:"pypi_max_file_size,omitempty"`
+	TerraformModuleMaxFileSize *int    `json:"terraform_module_max_file_size,omitempty"`
```
Comment (svanharmelen) [inline plan_limits.go:41]: These should all not be pointers and should be ordered alphabetically following the docs.

File: plan_limits.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	NugetMaxFileSize           *int   `json:"nuget_max_file_size,omitempty" url:"nuget_max_file_size,omitempty"`
+	PyPiMaxFileSize            *int   `json:"pypi_max_file_size,omitempty" url:"pypi_max_file_size,omitempty"`
+	TerraformModuleMaxFileSize *int   `json:"terraform_module_max_file_size,omitempty" url:"terraform_module_max_file_size,omitempty"`
+}
+
+// List the current limits of a plan on the GitLab instance.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/plan_limits.html#get-current-plan-limits
+func (s *PlanLimitsService) GetCurrentPlanLimits(options ...RequestOptionFunc) (*PlanLimit, *Response, error) {
```
Comment (svanharmelen) [inline plan_limits.go:62]: This one needs an options struct that takes a `PlanName`

File: plan_limits.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	PlanName                   string `json:"plan_name,omitempty" url:"plan_name,omitempty"`
+	ConanMaxFileSize           *int   `json:"conan_max_file_size,omitempty" url:"conan_max_file_size,omitempty"`
+	GenericPackagesMaxFileSize *int   `json:"generic_packages_max_file_size,omitempty" url:"generic_packages_max_file_size,omitempty"`
+	HelmMaxFileSize            *int   `json:"helm_max_file_size,omitempty" url:"helm_max_file_size,omitempty"`
+	MavenMaxFileSize           *int   `json:"maven_max_file_size,omitempty" url:"maven_max_file_size,omitempty"`
+	NpmMaxFileSize             *int   `json:"npm_max_file_size,omitempty" url:"npm_max_file_size,omitempty"`
+	NugetMaxFileSize           *int   `json:"nuget_max_file_size,omitempty" url:"nuget_max_file_size,omitempty"`
+	PyPiMaxFileSize            *int   `json:"pypi_max_file_size,omitempty" url:"pypi_max_file_size,omitempty"`
+	TerraformModuleMaxFileSize *int   `json:"terraform_module_max_file_size,omitempty" url:"terraform_module_max_file_size,omitempty"`
+}
```
Comment (svanharmelen) [inline plan_limits.go:56]: Option structs are made individually for a method. Please see the other code in this repo and follow the same pattern.

File: plan_limits.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	var planlimit *PlanLimit
+	resp, err := s.client.Do(req, &planlimit)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return planlimit, resp, err
+}
+
+// Modify the limits of a plan on the GitLab instance.
```
Comment (svanharmelen) [inline plan_limits.go:79]: This will give a linter warning

File: plan_limits.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	GenericPackagesMaxFileSize *int   `json:"generic_packages_max_file_size,omitempty" url:"generic_packages_max_file_size,omitempty"`
+	HelmMaxFileSize            *int   `json:"helm_max_file_size,omitempty" url:"helm_max_file_size,omitempty"`
+	MavenMaxFileSize           *int   `json:"maven_max_file_size,omitempty" url:"maven_max_file_size,omitempty"`
+	NpmMaxFileSize             *int   `json:"npm_max_file_size,omitempty" url:"npm_max_file_size,omitempty"`
+	NugetMaxFileSize           *int   `json:"nuget_max_file_size,omitempty" url:"nuget_max_file_size,omitempty"`
+	PyPiMaxFileSize            *int   `json:"pypi_max_file_size,omitempty" url:"pypi_max_file_size,omitempty"`
+	TerraformModuleMaxFileSize *int   `json:"terraform_module_max_file_size,omitempty" url:"terraform_module_max_file_size,omitempty"`
+}
+
+// List the current limits of a plan on the GitLab instance.
```
Comment (svanharmelen) [inline plan_limits.go:58]: This will give a linter warning.

**Review by svanharmelen — CHANGES_REQUESTED** [review]

**Comment by Urento** [comment]
I will fix these next week

**Pull request merged**